File: src/lib/types/exercise.ts
export interface ExerciseEntry {
	id: string;
	day_attempt_id: string;
	language: string;
	code: string;
	notes: string;
	order_index: number;
	created_at: string;
	updated_at: string;
}

export interface CreateExerciseEntryInput {
	day_attempt_id: string;
	language: string;
	code: string;
	notes: string;
}

export interface UpdateExerciseEntryInput {
	language?: string;
	code?: string;
	notes?: string;
}

File: src-tauri/src/db/models/exercise.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct ExerciseEntry {
    pub id: String,
    pub day_attempt_id: String,
    pub language: String,
    pub code: String,
    pub notes: String,
    pub order_index: i32,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateExerciseEntryInput {
    pub day_attempt_id: String,
    pub language: String,
    pub code: String,
    pub notes: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UpdateExerciseEntryInput {
    pub language: Option<String>,
    pub code: Option<String>,
    pub notes: Option<String>,
}
